.github/workflows/wheels.yml

```diff
@@ -4,63 +4,107 @@ on:
   workflow_dispatch: {}
 
 env:
-  # all values:
-  # 3.5;3.7;5.0;5.2;5.3;6.0;6.1;6.2;7.0;7.2;7.5;8.0;8.6
-  # with optional +PTX
-  # https://github.com/pytorch/vision/blob/d710f3d1edc06afa244468cb96603ba6dbd4d9d5/packaging/pkg_helpers.bash#L63
-  # you need at least cuda 5.0 for some of the stuff compiled here.
-  TORCH_CUDA_ARCH_LIST: "6.0 6.1 7.0 7.5 8.0 8.6"
   FORCE_CUDA: 1
+  MAX_JOBS: 1 # will crash otherwise
+  DISTUTILS_USE_SDK: 1 # otherwise distutils will complain on windows about multiple versions of msvc
 
 jobs:
   build_wheels:
     strategy:
+      fail-fast: false # don't stop other jobs if one failed
       matrix:
+        os:
+          - ubuntu-22.04
+          - windows-2019
         python:
-          - "python3.7"
-          - "python3.8"
-          - "python3.9"
-          - "python3.10"
+          - "3.7"
+          - "3.8"
+          - "3.9"
+          - "3.10"
         config:
           # torch 1.13 is built with cuda 11.6 and 11.7
           - torch_version: "1.13.0+cu117"
+            cuda_version: "11.7.0"
             pip_index: https://download.pytorch.org/whl/cu117
             cuda_run_file: https://developer.download.nvidia.com/compute/cuda/11.7.1/local_installers/cuda_11.7.1_515.65.01_linux.run
 
           - torch_version: "1.13.0+cu116"
+            cuda_version: "11.6.2"
             pip_index: https://download.pytorch.org/whl/cu116
             cuda_run_file: https://developer.download.nvidia.com/compute/cuda/11.6.2/local_installers/cuda_11.6.2_510.47.03_linux.run
 
           # torch 1.12 is built with cuda 11.3 and 11.6, we ignore 11.3 for now
           - torch_version: "1.12.1+cu116"
+            cuda_version: "11.6.2"
             pip_index: https://download.pytorch.org/whl/cu116
             cuda_run_file: https://developer.download.nvidia.com/compute/cuda/11.6.2/local_installers/cuda_11.6.2_510.47.03_linux.run
 
-    name: ${{ matrix.python }}-torch${{ matrix.config.torch_version }}
-    runs-on: ubuntu-22.04
+    name: ${{ matrix.os }}-py${{ matrix.python }}-torch${{ matrix.config.torch_version }}
+    runs-on: ${{ matrix.os }}
+    env:
+      # alias for the current python version
+      # windows does not have per version binary, it is just 'python3'
+      PY: python${{ contains(matrix.os, 'ubuntu') && matrix.python || '3' }}
+      # all values:
+      # 3.5;3.7;5.0;5.2;5.3;6.0;6.1;6.2;7.0;7.2;7.5;8.0;8.6
+      # with optional +PTX
+      # https://github.com/pytorch/vision/blob/d710f3d1edc06afa244468cb96603ba6dbd4d9d5/packaging/pkg_helpers.bash#L63
+      # you need at least cuda 5.0 for some of the stuff compiled here.
+      # windows machines will OOM with many architectures, hence they are reduced here
+      TORCH_CUDA_ARCH_LIST: ${{ contains(matrix.os, 'ubuntu') && '6.0 6.1 7.0 7.5 8.0 8.6' || '7.5 8.0 8.6' }}
```

Comment:
Can we get around using swap on windows?
It can be quite confusing for the user to only have a few arch supported on windows...
Something like this:
https://github.com/oktadev/auth0-full-stack-java-example/commit/580e96fb1c8f25d023f0f6d3e3d4535f3c3d3790

Reply:
sneaky! I like it.